bot/exts/halloween/spooknamerate.py

```diff
@@ -168,34 +172,37 @@ async def on_reaction_add(self, reaction: Reaction, user: User) -> None:
                     return await reaction.remove(user)  # removes user's reaction
 
     @spook_name_rate.command(name='delete')
-    async def delete_word(self, ctx: Context) -> None:
-        """Delete the user's word."""
+    async def delete_name(self, ctx: Context) -> None:
+        """Delete the user's name."""
         if self.poll:
-            await ctx.send("You can't delete your word since the poll has already started!")
+            await ctx.send("You can't delete your name since the poll has already started!")
             return
-        for message_id in self.messages:
-            data = self.messages[message_id]
-            if ctx.author == data['author']:
+        for message_id, data in self.messages.items():
+            if ctx.author.id == data['author']:
                 del self.messages[message_id]
-                await ctx.send(f'Message deleted successfully ({data["word"]!r})!')
+                self.update_json()
+                await ctx.send(f'Message deleted successfully ({data["name"]!r})!')
                 return
 
         await ctx.send("But you don't have an entry... :eyes: Type `spooknamerate add your entry`")
 
-    @tasks.loop(hours=24.0)
-    async def announce_word(self) -> None:
+    @tasks.loop(minutes=60.0)
+    async def announce_name(self) -> None:
         """Announces the name needed to spookify every 24 hours and the winner of the previous game."""
         if not self.in_allowed_month():
             return
+        if not datetime.utcnow().day == 12:  # Check if it is running at 12 PM, and only run it then
+            return
```

Comment:
The thought is correct, but I do not believe this is very robust. Also isn't .day refering to the date? not hour mark.
asyncio has logic to wait for a specific time, you could use that with a task that starts up wehn your bot does. There's probably others with more suggestions on how to do this

Reply:
Oops I think I commited some testing code